README.md

```diff
@@ -46,83 +114,77 @@ class Message:ObjectCoder {
 
-
-```
-> **MyNetworkClient** is a class implementing **NetworkInterface** (implementing **GET,PUT,POST,DELETE**) so that you can use any networking library of your choice
-
-# Create A CoreData Model
-> Make sure you have an xcdatamodel file named MyApp.xcdatamodeld in your project and has an entity "Message" defined with fields according to your Message.swift class(read above)
+Now you need is a Restify Model that can perform network calls on Message class. You can implement this in a Factory class so that you can easily make network calls on Message from anywhere in your project easily. 
 
 ```Swift
+        class ModelFactory {
 
-      static let coreDataStack = CoreDataStack(dbName: "MyApp")
-      let model = CoreDataStore<Message>(entityName: "Message", managedContext: coreDataStack.context)
-
+       	class func message() -> Restify<Message> {
+        	let messsageUrl = "http://api.mysite.com/messages"
+        	let model = Restify<Message>(path: messsageUrl, networkClient: MyNetworkClient()) // 	see note below to know about MyNetworkClient
+        	}
+}
 ```
-### your xcdatamodel should look something like this
-![logo](http://i.imgur.com/qNSIcTK.png?1)
-
-## Concept
-
-DataStoreKit works on two high level protocols 
+> **MyNetworkClient** is a class implementing **ModelProtocol** (implementing **GET,PUT,POST,DELETE**) so that you can use any networking library of your choice
 
-#### ModelProtocol
-
-ModelProtocol protocol is implemented by all the DataStores i.e. the CoreDataStore, Restify, UserDefaultStore etc.This provides a consistent api to access your data.
+Now, you are ready to make calls to your server and fetch data. You can now perform following requests to your server:- 
 
 ```Swift
+	   let paramsDic: [String:Any]? = [“from”: “1234567890”, “to”:”0987654321”]
+	   ModelFactory.messages().query(params: paramsDic, options: nil, callback: { (error:NSError?, results:Any?) in
+         // Do with results     
+       })
+```
 
 
-public protocol ModelProtocol:class {
-    
-    func query(params params:[String:AnyObject]?, options:[String:AnyObject]?, callback: ModelArrayCallback? )
-    func all(callback:ModelArrayCallback?)
-    func get(id id:String?,params:[String:AnyObject]?, callback: ModelObjectCallback? )
-    func put(id id:String?,object:ObjectCoder, callback: ModelObjectCallback? )
-    func add(object:ObjectCoder, callback: ModelObjectCallback? )
-    func remove(id id:String?,params:[String:AnyObject]?, callback: ModelObjectCallback? )
+The result of above network call will be array of objects of Message class that contains messages with **from as 1234567890** and **to as 0987654321**
 
-}
-``` 
+Above code is for QUERYING Data between two users. Similarly you can use GET, PUT, ADD, REMOVE. 
 
-```Swift
-public typealias ModelArrayCallback = (NSError?,NSArray?)->Void
-public typealias ModelObjectCallback = (NSError?,AnyObject?)->Void
-```
+#### Local Storage Calls 
```

Comment:
give examples of how to query with less than and greater than operations in core data.

All the cases which are in Unit Tests should come under Readme